src/clds_sorted_list.c

```diff
@@ -769,8 +769,11 @@ CLDS_SORTED_LIST_INSERT_RESULT clds_sorted_list_insert(CLDS_SORTED_LIST_HANDLE c
                 // get the current_item value
                 volatile CLDS_SORTED_LIST_ITEM* current_item = (volatile CLDS_SORTED_LIST_ITEM*)InterlockedCompareExchangePointer((volatile PVOID*)current_item_address, NULL, NULL);
 
+                // clear any delete lock bit from what we read
+                current_item = (volatile CLDS_SORTED_LIST_ITEM*)((uintptr_t)current_item & ~0x1);
```

Comment:
>current_item = (volatile CLDS_SORTED_LIST_ITEM*)((uintptr_t)current_item & ~0x1); [](start = 16, length = 81)

I kind of feel like this should be wrapped in some kind of handle/macros to make it less likely to miss...

Ideally, the "handle" would expose something like set/clear bit and "get pointer without delete bit" and somehow fail if it's used incorrectly.

maybe not worth doing right now but could be a task for the backlog.

Reply:
Yeah, I guess at a later time ....

---
In reply to: [421122049](https://github.com/dcristoloveanu/clds/pull/47#discussion_r421122049) [](ancestors = 421122049)